commit de52140fe4e80bc5dd0ad672e4f9d792e0868735
Author: Brian Bennett <brian.bennett@joyent.com>
Date:   2019-03-18T18:44:05-07:00 (7 months ago)
    
    TRITON-599 CNS needs better handling of acme challenges for new instances

diff --git a/lib/vm-to-zones.js b/lib/vm-to-zones.js
index 3ac9507..cffe2a7 100644
--- a/lib/vm-to-zones.js
+++ b/lib/vm-to-zones.js
@@ -36,17 +36,15 @@ function buildZonesFromVm(vm, config, log) {
 						network: nic.network
 					});
 				}
-				if (vm.listServices) {
-					vm.services.forEach(function (svc) {
-						entries.push({
-							type: 'service',
-							ip: ip,
-							zone: zone,
-							service: svc,
-							network: nic.network
-						});
+				vm.services.forEach(function (svc) {
+					entries.push({
+						type: 'service',
+						ip: ip,
+						zone: zone,
+						service: svc,
+						network: nic.network
 					});
-				}
+				});
 			});
 		});
 	});
@@ -276,17 +274,19 @@ function addService(zones, vm, ent, config) {
 		if (!zones[ent.zone][name])
 			zones[ent.zone][name] = [];
 		var recs = zones[ent.zone][name];
-		recs.push({
-			constructor: ent.addrType,
-			args: [ent.ip],
-			src: vm.uuid
-		});
 		var hasTxt = false;
-		for (var i = 0; i < recs.length; ++i) {
-			if (recs[i].constructor === 'TXT' &&
-			    recs[i].args[0] === vm.uuid) {
-				hasTxt = true;
-				break;
+		if (vm.listServices) {
+			recs.push({
+				constructor: ent.addrType,
+				args: [ent.ip],
+				src: vm.uuid
+			});
+			for (var i = 0; i < recs.length; ++i) {
+				if (recs[i].constructor === 'TXT' &&
+				    recs[i].args[0] === vm.uuid) {
+					hasTxt = true;
+					break;
+				}
 			}
 		}
 		if (!hasTxt) {
@@ -298,17 +298,19 @@ function addService(zones, vm, ent, config) {
 		}
 	}
 	function addSRV(name, port) {
-		if (!zones[ent.zone])
-			zones[ent.zone] = {};
-		if (!zones[ent.zone][name])
-			zones[ent.zone][name] = [];
-		var recs = zones[ent.zone][name];
-		var target = primaryName(vm, config) + '.' + ent.zone;
-		recs.push({
-			constructor: 'SRV',
-			args: [target, port],
-			src: vm.uuid
-		});
+		if (vm.listServices) {
+			if (!zones[ent.zone])
+				zones[ent.zone] = {};
+			if (!zones[ent.zone][name])
+				zones[ent.zone][name] = [];
+			var recs = zones[ent.zone][name];
+			var target = primaryName(vm, config) + '.' + ent.zone;
+			recs.push({
+				constructor: 'SRV',
+				args: [target, port],
+				src: vm.uuid
+			});
+		}
 	}
 
 	function addACME(name) {
